examples/call_watch/src/main.rs

```diff
@@ -1,53 +1,46 @@
 // Example of watching for new DAI transfers
 // WARNING: This example doesn't account for rollbacks
 
-use hypersync_client::{net_types::Query, CallDecoder, Client, ClientConfig};
+use hypersync_client::{
+    net_types::{Query, TransactionField, TransactionFilter},
+    CallDecoder, Client, ClientConfig,
+};
 use tokio::time::{sleep, Duration};
 
 const DAI_ADDRESS: &str = "0x6B175474E89094C44Da98b954EedeAC495271d0F";
 
 #[tokio::main]
-async fn main() {
+async fn main() -> anyhow::Result<()> {
     env_logger::init().unwrap();
 
     // create default client, uses eth mainnet
     let client = Client::new(ClientConfig::default()).unwrap();
 
     // The query to run
-    let mut query: Query = serde_json::from_value(serde_json::json!( {
+    let mut query = Query::new()
         // start from tip since we only want new transfers
-        "from_block": 20519993,
-        // The logs we want. We will also automatically get transactions and blocks relating to these logs (the query implicitly joins them).
-        "transactions": [
-            {
-                "from": [DAI_ADDRESS],
-            },
-            {
-                "to": [DAI_ADDRESS],
-            },
-        ],
+        .from_block(20519993)
+        .where_transactions(
+            // The logs we want. We will also automatically get transactions and blocks relating to these logs (the query implicitly joins them).
+            // We want all DAI transfers so no address filter and only a filter for the first topic
+            TransactionFilter::all()
+                .and_from_address([DAI_ADDRESS])?
+                .and_to_address([DAI_ADDRESS])?,
```

Comment:
Could be named just `and_to` `and_from`, but happy with more explicit 👍

Reply:
Can do, but my reasoning was that it's not explicit what type needs to be passed into the function. 

Probably you're right though. For eg. if we got a new field on transactions table called "from_address" instead of "from" then this wouldn't namespace nicely

Reply:
@JasoonS I've done this in the latest commit. I agree 👍🏼